shared/src/main/scala/coop/rchain/shared/EventLog.scala

```diff
@@ -0,0 +1,26 @@
+package coop.rchain.shared
+
+sealed trait Event
+object Event {
+  final case class NodeStarted(address: String)                             extends Event
+  final case class EnteredRunningState(blockHash: String)                   extends Event
+  final case class ApprovedBlockReceived(blockHash: String)                 extends Event
+  final case class SentUnapprovedBlock(blockHash: String)                   extends Event
+  final case class BlockApprovalReceived(blockHash: String, sender: String) extends Event
+  final case class SentApprovedBlock(blockHash: String)                     extends Event
+}
+
+trait EventLog[F[_]] {
+  def publish(event: Event): F[Unit]
+}
+
+object EventLog {
+  def apply[F[_]](implicit E: EventLog[F]): EventLog[F] = E
+
+  implicit val EventLoggerLogSource: LogSource = LogSource(classOf[EventLogger])
+
+  def eventLogger[F[_]: Log]: EventLog[F] =
+    (event: Event) => Log[F].info(event.getClass.getSimpleName.stripSuffix("$"))
```

Comment:
oh it is used, just generically! We should probably do a pattern match in here, without a 'catch-all' case so that we have the compiler yell at us on new unsupported event types.

Also, we should be leveraging the fields you've just added

Reply:
What format do you propose? If it's Json then we need some further specification, f.e.
```json
{
  "name": "BlockApprovalReceived",
  "ts": 12345,
  "body": {
    "block-hash": "abb989aabcc"
  }
}
```

Reply:
I'd say the specification should be 'dump the event case class structure into JSON generically, that's the JSON the output specification defines' :joy: 

If we're not happy with the JSON structure thus achieved, we should probably not be happy with the event case classes either :)